src/main/java/Duke/Ui.java

```diff
@@ -0,0 +1,144 @@
+package Duke;
+
+import Duke.Exception.*;
+
+import java.io.InputStream;
+import java.io.PrintStream;
+import java.util.Scanner;
+
+/**
+ * Ui deals with interactions with the user.
+ */
+public class Ui {
+
+    /**
+     * UI line divider
+     */
+    public static final String DIVIDER = "------------------------------------------------\n";
+
+    /**
+     * Introduction message
+     */
+    public static final String GREET =
+            " ____        _        \n"
+                    + "|  _ \\ _   _| | _____ \n"
+                    + "| | | | | | | |/ / _ \\\n"
+                    + "| |_| | |_| |   <  __/\n"
+                    + "|____/ \\__,_|_|\\_\\___|\n"
+                    + DIVIDER
+                    + "Hello! I'm Duke\nWhat can I do for you?\n";
+
+    /**
+     *  Farewell message
+     */
+    public static final String FAREWELL = "Bye. Hope to see you again soon!\n";
+
+    private static final String LS = System.lineSeparator();
+
+    private final Scanner in;
+    private final PrintStream out;
+
+    public Ui() {
+        this(System.in, System.out);
+    }
+
+    public Ui(InputStream in, PrintStream out) {
```

Comment:
Why does this overloaded method needs to be public if it's only used inside this class ?